cpp/arcticdb/storage/config_cache.hpp

```diff
@@ -80,8 +80,19 @@ class ConfigCache {
         util::check(!descriptor.storage_ids_.empty(), "Can't configure library with no storage ids");
         std::vector<std::unique_ptr<VariantStorage>> variants;
         for (const auto& storage_name : descriptor.storage_ids_) {
-            auto factory = get_storage_factory(storage_name);
-            variants.emplace_back(factory->create_storage(path, mode));
+            // Otherwise see if we have the storage config.
+            arcticdb::proto::storage::VariantStorage storage_conf;
+            auto storage_conf_pos = storage_configs_.find(storage_name);
+            if(storage_conf_pos != storage_configs_.end())
+                storage_conf = storage_conf_pos->second;
+
+            // As a last resort, get the whole environment config from the resolver.
+            refresh_config();
+            storage_conf_pos = storage_configs_.find(storage_name);
+            if(storage_conf_pos != storage_configs_.end())
+                storage_conf = storage_conf_pos->second;
+
+            variants.emplace_back(create_storage(path, mode, storage_conf));
```

Comment:
Note for reviewers: those lines simply haved moved from `get_storage_factory_internal` which has been deleted.

Reply:
That's cool. We don't use the file config much, but it's not a totally bad feature to have, and this addresses the situation where a config has been added after the config file has been loaded.